src/Components/Facility/FacilityCard.tsx

```diff
@@ -14,26 +11,28 @@ import DialogModal from "../Common/Dialog";
 import TextAreaFormField from "../Form/FormFields/TextAreaFormField";
 import useConfig from "../../Common/hooks/useConfig";
 import { classNames } from "../../Utils/utils";
+import request from "../../Utils/request/request";
+import routes from "../../Redux/api";
 
 export const FacilityCard = (props: { facility: any; userType: any }) => {
   const { facility, userType } = props;
   const { kasp_string } = useConfig();
 
   const { t } = useTranslation();
-  const dispatchAction: any = useDispatch();
   const [notifyModalFor, setNotifyModalFor] = useState(undefined);
   const [notifyMessage, setNotifyMessage] = useState("");
   const [notifyError, setNotifyError] = useState("");
 
   const handleNotifySubmit = async (id: any) => {
-    const data = {
-      facility: id,
-      message: notifyMessage,
-    };
-    if (data.message.trim().length >= 1) {
+    if (notifyMessage.trim().length >= 1) {
       setNotifyError("");
-      const res = await dispatchAction(sendNotificationMessages(data));
-      if (res && res.status == 204) {
+      const { res } = await request(routes.sendNotificationMessages, {
+        body: {
+          facility: id,
+          message: notifyMessage,
+        },
+      });
+      if (res?.ok && res.status == 204) {
```

Comment:
`res.status == 204` is not required if you are doing `res?.ok` as res.ok checks if the status code is a success status code.

```suggestion
      if (res?.ok) {
```